packages/shopify-api/lib/fulfillment-service/validate.ts

```diff
@@ -0,0 +1,64 @@
+import {abstractConvertRequest, getHeader} from '../../runtime/http';
+import {HashFormat} from '../../runtime/crypto/types';
+import {ConfigInterface} from '../base-types';
+import {logger} from '../logger';
+import {ShopifyHeader} from '../types';
+import {validateHmacString} from '../utils/hmac-validator';
+
+import {
+  FulfillmentServiceErrorReason,
+  FulfillmentServiceInvalid,
+  FulfillmentServiceParams,
+  FulfillmentServiceValid,
+} from './types';
+
+export function validateFactory(config: ConfigInterface) {
+  return async function validate({
```

Comment:
As per above, this could be a shared bit of code that takes in a body + request combo and does the validation.